### PR TITLE
Fixes for macOS

### PR DESCRIPTION
**Pull request by athei**
- Don't assume python path
- Minimize config by using "user" profile
- Disable `split-debuginfo` (errors out on macOS)
- Dynamically determine host toolchain triplet
- Link stage2 and not stage1 (stage1 misses target libs for me

`-Z build-std=core,alloc` still fails for me with this toolchain. We should check again later when we update to latest rust. Maybe it will be fixed.

### CODE REVIEW

File: 05-build-dist.sh [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 set -euo pipefail
 
+source config.sh
+
 cd rust
 mkdir -p build/riscv32em-unknown-none-elf/compiler-doc
-python x.py dist
+./x dist --target $TOOLCHAIN_HOST_TRIPLET --target riscv32em-unknown-none-elf
```
Comment (gab8i) [inline 05-build-dist.sh:9]: Does the first target flag compile the dist for the specified target? Is this correct? Could I then use this flag to compile multiple dists, one for linux and one for macOS (maybe even one for windows) on the same machine?

Is the second flag instead specified because you removed it from the config.toml? If yes, why did you move it here?

Comment (koute) [reply 05-build-dist.sh:9]: > Does the first target flag compile the dist for the specified target? Is this correct? Could I then use this flag to compile multiple dists, one for linux and one for macOS (maybe even one for windows) on the same machine?

This tells the build system for which target systems the resulting Rust compiler should be able to emit binaries for, and *not* for which systems to build the toolchain to run *on*.

In practice you usually cannot (easily) compile the toolchain so that it *runs* on a different system than the one on which you are compiling.

It should be possible to cross-compile it in certain cases, but that requires a cross-compiler to be installed, which is often tricky. (Especially for macOS, because the full toolchain/libraries required are proprietary.)

> Is the second flag instead specified because you removed it from the config.toml? If yes, why did you move it here?

Yes. It was moved so that one of the target triplets can be set to be the same as the host system.

Basically, our Rust toolchain must be able to target two targets:

1) the RISC-V target (`riscv32em-unknown-none-elf`, so that it can compile guest programs for the VM)
2) the host target (so that it can compile procedural macros which the guest programs might use at compile time)

Comment (gab8i) [reply 05-build-dist.sh:9]: > In practice you usually cannot (easily) compile the toolchain so that it runs on a different system than the one on which you are compiling.

Ok, perfect, thanks!

> Yes. It was moved so that one of the target triplets can be set to be the same as the host system.

Ohh, I see now. @athei modified the `config.toml` file changing the profile and then removing the targets and making it dynamic using `config.sh`. Is my understanding correct? I want to make sure I understand the changes properly before I start experimenting with macOS distribution creation.

Comment (athei) [reply 05-build-dist.sh:9]: Yes. Exactly. Before it was hardcoded to a linux target which would not be of much use when compiling on a macOS machine.

**Comment by athei** [comment]
Toolchain for macOS can be found here:

https://github.com/paritytech/rustc-rv32e-toolchain
